haystack/nodes/summarizer/base.py

```diff
@@ -14,43 +14,34 @@ class BaseSummarizer(BaseComponent):
     outgoing_edges = 1
 
     @abstractmethod
-    def predict(self, documents: List[Document], generate_single_summary: Optional[bool] = None) -> List[Document]:
+    def predict(self, documents: List[Document]) -> List[Document]:
         """
         Abstract method for creating a summary.
 
         :param documents: Related documents (e.g. coming from a retriever) that the answer shall be conditioned on.
-        :param generate_single_summary: This parameter is deprecated and will be removed in Haystack 1.12
         :return: List of Documents, where Document.meta["summary"] contains the summarization
         """
         pass
 
     @abstractmethod
     def predict_batch(
-        self,
-        documents: Union[List[Document], List[List[Document]]],
-        generate_single_summary: Optional[bool] = None,
-        batch_size: Optional[int] = None,
+        self, documents: Union[List[Document], List[List[Document]]], batch_size: Optional[int] = None
     ) -> Union[List[Document], List[List[Document]]]:
         pass
 
-    def run(self, documents: List[Document], generate_single_summary: Optional[bool] = None):  # type: ignore
+    def run(self, documents: List[Document]):  # type: ignore
```

Comment:
Do we still need `# type: ignore` here?

Reply:
Unfortunately yes.
Otherwise, mypy complains:
`haystack/nodes/summarizer/base.py:32: error: Signature of "run" incompatible with supertype "BaseComponent"  [override]`

Reply:
Cool. Then let's merge this! :+1: